### PR TITLE
Refactoring login tests. Adding new test cases 

### PR DESCRIPTION
**Pull request by denik1981**
## What does this PR do?

This is part of a series of PRs whose goal is to optimize all the tests, extend the whole test coverage, and finally improve the inner test running performance. These tests are all using the new fixture addition which by the end of all these series will allow us to run tests concurrently and speed them up in consequence.  

A summary of changes: 

- New test case: login for each type of account (plan) and asserting for the plan info as well.
- New test case: login for a non-existent user account.
- New test case: login with a wrong password.
- New test case: logout using the user's expected flow (using the dropdown)
- Add assertive declarations missing in the previous version for login tests. 
- Removing unnecessary types in the user's fixture that adds up to unwanted boilerplate (type inference when possible).
- Adding a small localize utility for tests that emulates what next-i18n is doing in the app. 

## Type of change
- [x] New feature (non-breaking change which adds functionality)

## How should this be tested?

- `yarn workspace @calcom/prisma db-reset`
- `yarn dev`
- `yarn workspace @calcom/web test-e2e`

### CODE REVIEW

File: apps/web/playwright/fixtures/types.ts [hunk 1]
```diff
@@ -0,0 +1,4 @@
+export enum TimeZoneE {
```
Comment (alannnc) [inline apps/web/playwright/fixtures/types.ts:1]: rename to `TimeZoneEnum` to maintain the context on its name

File: apps/web/playwright/fixtures/users.ts [hunk 2]
```diff
@@ -82,15 +67,22 @@ const createUser = async (opts?: CustomUserOpts) => {
...
-  await page.fill('input[name="email"]', user.email);
-  await page.press('input[name="email"]', "Tab");
-  await page.fill('input[name="password"]', user.username!);
-  await page.press('input[name="password"]', "Enter");
+  await emailLocator.fill(user.email ?? `${user.username}@example.com`);
+  await passwordLocator.fill(user.password ?? user.username!);
+  await signInLocator.click();
 
-  // 2 seconds of delay before returning to help the session loading well
+  //2s of delay to give the session enough time for a clean load
```
Comment (alannnc) [inline apps/web/playwright/fixtures/users.ts:86]: space after comment // 2 seconds

**Review by alannnc — DISMISSED** [review]
NIT changes, but test seem to be working fine

File: apps/web/playwright/fixtures/users.ts (hunk 2 above)
Comment (denik1981) [reply apps/web/playwright/fixtures/users.ts:86]: done!

**Comment by denik1981** [comment]
> NIT changes, but test seem to be working fine

Thanks for reviewing this!

**Review by zomars — APPROVED** [review]
Nice work!

**Comment by zomars** [comment]
@denik1981 for some reason since this got merged, test are now timing out at the 10 minute mark on GH. Can you help us debug this? Also should we add configs to the `e2e.yml` file?

**Comment by denik1981** [comment]
It works well on my side (locally) but let me check the public logs in the PRs I have available. 
Retrials maybe? 

One easy and fast fix is to reduce the timeout for each retrial. Currently, it is set to 60 seconds to wait for the test to acknowledge it is not working and retry. This could be reduced to half at least (and even further IMHO). But 2 or 3 retrials can eat more than 3 minutes from the total time. 

The other thing is something I cannot do anything about it in the short run which is that we are running with only one worker when Playwright is planned to be run multithreaded but this was necessary for us because we are using the same source of data for each test and it was the cause some of them were failing.

I'm working on this. In replacing everything we have with fixtures that should give us more power to test everything in parallel. My current work is with the "schedules" fixtures that is a dependency on the "events" fixture. I will push this one early tomorrow with some fixes I found as well for some issues related to availability and a set of tests for the entire availability section of Cal.  And the last picture remaining will be the "teams" fixture so that I can replace all the calls to the seeded data and raise the number of workers to an optimal number that can speed up our tests. Booking is another one but it can be deferred for later as it is not stopping me from migrating tests to the new schema.   

The truth is that the more tests we add, the more time we will spend running those tests because they are all running in serial mode. This was something necessary to do but far from ideal in terms of running our tests in CI.

Even do, I will check the logs and see where is the source of the problem to help you fix it.

**Comment by denik1981** [comment]
Unfortunately, I couldn't find any run in the public logs using the login/logout tests addition but I pulled an updated version of Cal with the latest PRs and run a CI test session locally to measure everything. I know is not the same but it help me issue a preflight research. Logging tests are taken 49s (locally) in total, ran without any issue, and actually very fast. 

I found a few retrials and, remember, each of them is eating 60 seconds. Some of these retrials came from failing tests which ended up eating 2 minutes per test in total. There is a very recent PR (#2239) that is adding tests for dynamic bookings and those tests fall into the group of failing tests. We might wanna check those tests and eventually mark them with `fixme` so that they don't run in CI for the moment until every issue in there is addressed.

There is also a recurrent test case that usually fails in the first run which is the booking reschedule. Not sure, what's going on with this test but it usually needs a second shot to run well. It might be good to mark this one with `fixme` as well.

Regarding `e2e.yml` that I didn't answer before, no, there is no change needed there (from my side at least) for the added login/logout tests. In fact, none of the other tests are compromised with any addition made in the login/logout tests. The users' fixture is only used in there and is planned to be used everywhere shortly. The rest of the tests are using seeded data as usual. 
I forgot to mention that an extra benefit of not using seeded data is that we can cut in the time needed to seed the DB as a pre-step for each e2e run. 

Below is a snapshot of my latest run locally of the e2e tests.

 - Blue: skipped 
 - Gray: Run (time measurements in brackets)
 - Red: failures
 
 
image

image


These (below) are all the login/logout tests. Less than 50 seconds in total. 
I predict this could be reduced to about ~15/20 seconds once we raise the amount of workers available.  

image

### SUMMARY
My suggestion is to mark with `fixme` the current failing tests and reduce the global timeout from the playwright configuration to as much as 30s if not less.  Login/out tests seem to be running well and fast. Even if raising workers is not possible now these tests can be further optimized (very subtle) because they are forcing a 2s timeout between each new login to give extra time for the session to prepare itself without running into troubles. I found that without these little gaps of time, login tests can fail, but 2s is a completely random value. I will try with only 1 second to see if it runs well. All in all, I don't think this will reduce the overall run for these tests in more than 10/15 seconds and I don't think is the reason of CI taking so long.

**Comment by denik1981** [comment]
Further on this:

I ran a bisect test to ultimately find the commit that is pulling down a lot of e2e tests. Note that I clean up the entire repo and reinstalled everything between these tests, and also rebuild the DB. 

So I first checked into the first commit that came after the commits that merge the Refactoring of Login/logout tests and everything is running well. Output below:

image
image



Then I made a second run with the next commit in the stack "dynamic group links (#2239)" which lead to a completely different output on where some tests are failing. If there is some extra configuration needed to test this commit please let me know because I'm making the (maybe wrong) assumption that it can be run as it is.

image
image
image


### Aside Note
From both runs, I can see the below test (which I mentioned before) fails in the first attempt. I'm not sure why this happens with this test but is happening since it has been there and the problem is that it demands to use the entire timeout (1 minute) before it fails and retries. In the second snapshot, this test fails in the retry as well with the added cost of eating 1 minute for a second time. We might want to investigate the source of the issue with this test and refactor it if possible, 

`[chromium] › ../../apps/web/playwright/booking-pages.test.ts:129:3 › pro user › can reschedule a booking`

**Pull request merged**